cpp/src/lance/io/reader.cc

```diff
@@ -297,6 +297,10 @@ ::arrow::Result<std::shared_ptr<::arrow::Array>> FileReader::GetArray(
     storage_arr = GetDictionaryArray(field, batch_id, params);
   } else {
     storage_arr = GetPrimitiveArray(field, batch_id, params);
+    if (!storage_arr.ok()) {
+      return storage_arr.status();
```

Comment:
how come in #104 we return `storage_arr` and here we return `storage_arr.status()` ?

Reply:
`storage_arr` is a `Result<Array>`. While return `storage_arr` has the same effect as returning `storage_arr.status()`. Here is for more clarity w.r.t. code reading. 


Both ways are fine to me tho.